### PR TITLE
Datashader lock files

### PR DESCRIPTION
**Pull request by kebowen730**
This PR adds lockfiles for the datashader based examples

### CODE REVIEW

**Comment by kebowen730** [comment]
@jbednar they are currently building and running for Mac and Linux, I used
'anaconda-project run' before locking the files. However, the walker_lake and seattle_lidar projects do seem to be running into issues, when I try to run or prepare them they stall out while trying to solve the environment. I'm not sure what's causing it as no error ever appears, unless they just have very long setup times.

**Comment by jbednar** [comment]
The key test is that after locking they still run, because once there is a lock file, that's the file that will be used in any subsequent run.  So once it's locked, it needs testing on each platform _with_the_lock_file_in_place_. Also, this change affects the .zip files that are created for when you download the project, and we'll need to document this both for users (who should be told to delete the lock file if they want to update the versions) and for maintainers (who should be told how to update the lock file).

File: bay_trimesh/anaconda-project.yml [hunk 1]
```diff
@@ -2,37 +2,37 @@
...
+- philippjfr
 labels:
-  - geoviews
-  - datashader
+- geoviews
+- datashader
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline bay_trimesh/anaconda-project.yml:11]:
```suggestion
user_fields: [labels, skip, maintainers]
```

Is there a bug in anaconda-project that is creating all those bogus copies of `user_fields` in this list?

File: attractors/anaconda-project.yml [hunk 2]
```diff
@@ -2,28 +2,29 @@
...
+- datashader
+- panel
 
 channels:
-  - pyviz
+- pyviz
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline attractors/anaconda-project.yml:14]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: carbon_flux/anaconda-project.yml [hunk 3]
```diff
@@ -1,55 +1,60 @@
...
+- jbednar
 labels:
-  - datashader
-  - geoviews
+- datashader
+- geoviews
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields, user_fields, user_fields]
```
Comment (jbednar) [inline carbon_flux/anaconda-project.yml:12]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: census/anaconda-project.yml [hunk 4]
```diff
@@ -2,40 +2,40 @@
...
+- jbednar
 labels:
-  - datashader
-  - geoviews
+- datashader
+- geoviews
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline census/anaconda-project.yml:11]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: datashader_dashboard/anaconda-project.yml [hunk 5]
```diff
@@ -1,37 +1,38 @@
...
-  - channel_conda-forge
-  - datashader
-  - panel
+- channel_conda-forge
+- datashader
+- panel
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline datashader_dashboard/anaconda-project.yml:15]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: datashader_dashboard/anaconda-project.yml [hunk 6]
```diff
@@ -1,37 +1,38 @@
 # To reproduce: install 'anaconda-project', then 'anaconda-project run'
 name: datashader_dashboard
-description: Interactive dashboard for making Datashader plots from any dataset that has latitude and longitude
+description: Interactive dashboard for making Datashader plots from any dataset that
+  has latitude and longitude
 created: 2018-11-12
 maintainers:
-  - jsignell
-  - jbednar
+- jsignell
```
Comment (jbednar) [inline datashader_dashboard/anaconda-project.yml:7]:
```suggestion
```

File: glaciers/anaconda-project.yml [hunk 7]
```diff
@@ -2,21 +2,21 @@
...
-  - panel
-  - geoviews
-  - datashader
+- panel
+- geoviews
+- datashader
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline glaciers/anaconda-project.yml:12]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: gull_tracking/anaconda-project.yml [hunk 8]
```diff
@@ -2,36 +2,36 @@
...
+- jbednar
 labels:
-  - datashader
-  - holoviews
+- datashader
+- holoviews
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline gull_tracking/anaconda-project.yml:11]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: heat_and_trees/anaconda-project.yml [hunk 9]
```diff
@@ -6,52 +6,52 @@ description: |
...
-  - hvplot
-  - geoviews
-  - datashader
+- hvplot
+- geoviews
+- datashader
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline heat_and_trees/anaconda-project.yml:16]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: iex_trading/anaconda-project.yml [hunk 10]
```diff
@@ -3,36 +3,35 @@
...
-  - panel
-  - holoviews
-  - dev
-  - channel_pyviz-dev
-  
+- datashader
+- panel
+- holoviews
+- dev
+- channel_pyviz-dev
```
Comment (jbednar) [inline iex_trading/anaconda-project.yml:12]: @jlstevens, the dev label can now be removed, right? And the dev channel below, and bump hv up to 1.13? If so please do so here.

File: landsat/anaconda-project.yml [hunk 11]
```diff
@@ -2,55 +2,59 @@
...
+- jbednar
 labels:
-  - geoviews
-  - datashader
+- geoviews
+- datashader
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline landsat/anaconda-project.yml:11]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: landsat_clustering/anaconda-project.yml [hunk 12]
```diff
@@ -1,56 +1,60 @@
 # To reproduce: install 'anaconda-project', then 'anaconda-project run'
 name: landsat_clustering
-description: Example of spectral clustering landsat data 
+description: Example of spectral clustering landsat data
```
Comment (jbednar) [inline landsat_clustering/anaconda-project.yml:3]:
```suggestion
description: Example of spectral clustering LANDSAT data
```

File: landsat_clustering/anaconda-project.yml [hunk 13]
```diff
@@ -1,56 +1,60 @@
...
+- jbednar
 labels:
-  - datashader
-  - geoviews
+- datashader
+- geoviews
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields, user_fields, user_fields, user_fields, user_fields]
```
Comment (jbednar) [inline landsat_clustering/anaconda-project.yml:11]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: landuse_classification/anaconda-project.yml [hunk 14]
```diff
@@ -2,50 +2,54 @@
...
+- jbednar
 labels:
-  - datashader
-  - geoviews
+- datashader
+- geoviews
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline landuse_classification/anaconda-project.yml:11]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: network_packets/anaconda-project.yml [hunk 15]
```diff
@@ -2,29 +2,29 @@
...
-  - holoviews
-  - datashader
-  - networkx
+- holoviews
+- datashader
+- networkx
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline network_packets/anaconda-project.yml:12]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: nyc_taxi/anaconda-project.yml [hunk 16]
```diff
@@ -2,29 +2,29 @@
...
+- jbednar
 labels:
-  - datashader
-  - panel
+- datashader
+- panel
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields, user_fields, user_fields, user_fields]
```
Comment (jbednar) [inline nyc_taxi/anaconda-project.yml:11]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: osm/anaconda-project.yml [hunk 17]
```diff
@@ -2,25 +2,25 @@
...
 maintainers:
-  - jbednar
+- jbednar
 labels:
-  - datashader
+- datashader
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields]
```
Comment (jbednar) [inline osm/anaconda-project.yml:10]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: uk_researchers/anaconda-project.yml [hunk 18]
```diff
@@ -2,57 +2,61 @@
...
 maintainers:
-  - philippjfr
+- philippjfr
 labels:
-  - datashader
+- datashader
 
-user_fields: [labels, skip, maintainers]
+user_fields: [labels, skip, maintainers, user_fields, user_fields, user_fields, user_fields,
+  user_fields, user_fields, user_fields]
```
Comment (jbednar) [inline uk_researchers/anaconda-project.yml:10]:
```suggestion
user_fields: [labels, skip, maintainers]
```

File: uk_researchers/anaconda-project.yml [hunk 19]
```diff
@@ -2,57 +2,61 @@
 name: uk_researchers
 description: UK research networks with HoloViews, Bokeh, and Datashader
 maintainers:
-  - philippjfr
+- philippjfr
```
Comment (jbednar) [inline uk_researchers/anaconda-project.yml:5]:
```suggestion
- philippjfr
- jbednar
```

File: bay_trimesh/anaconda-project.yml (hunk 1 above)
Comment (kebowen730) [reply bay_trimesh/anaconda-project.yml:11]: I believe there is, I'll go through and clean up all the extra `user_fields` entries.

**Comment by kebowen730** [comment]
I reverted my changes to seattle_lidar because while I was able to generate a lock file, when I try to run it I run into this ImportError with libffi:
`ImportError: dlopen(/Users/kbowen/pyviz-topics/examples/seattle_lidar/envs/default/lib/python3.6/lib-dynload/_ctypes.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libffi.6.dylib
  Referenced from: /Users/kbowen/pyviz-topics/examples/seattle_lidar/envs/default/lib/python3.6/lib-dynload/_ctypes.cpython-36m-darwin.so
  Reason: image not found`

I've tried updating or installing libffi with `conda install libffi` and `brew install libffi`, neither of which have yielded results.
I'm running into the same issue with walker_lake. I'm going to go ahead and merge the other lock files, and then continue looking for a solution for these two.

**Pull request merged**